### PR TITLE
Feature/uc3

### PR DESCRIPTION
**Pull request by MishkaMN**
# PR Details
## Description
The end result of this PR is that carma-platform is able to behave like UC2 by processing MAP/SPAT sent with a similar structure of UC3 (ignoring the strategy parameters). Fixes and implementations to achieve that is as follows:

**carma_wm_ctrl:**
- Fixed an issue where carma-platform was not able to process multiple signal_groups in a single entry lane (the linked issue). 
- Updated carma_wm_ctrl's MAP msg offset for intersections with most recent message from carma-street

**carma_wm:**
- Fixed an issue where carma_wm was not updating actual lanelet objects when processing dynamic SPAT (might be easier to only compare the changes in ROS1 as ROS2 was entirely missing the logic, so it looks big).
- Matched every carma_wm's ROS1 code with ROS2.
- removed outdated logic https://github.com/usdot-fhwa-stol/carma-platform/issues/1553

**arbitrator:**
- Fixed (temporary) an issue where lci_strategic_plugin's namespace was no correctly picked up by arbitrator

**lci_tactical_plugin:**
- Fixed light_controlled_intersection_tactical_plugin had a bug from porting to ROS2

**lci_strategic_plugin:**
- Fixed all the instances where the logic assumes the signal states are fixed, and handles them appropriately for dynamic states. End result now is that the vehicle can behave like UC2 (estimate its own entry time based on signal states) by processing MAP/SPAT sent with a similar structure of UC3 (ignoring the strategy parameters). 
- Future implementation is needed to fully support UC3, once carma-street is able to send strategy_params. Issue is created here:
https://github.com/usdot-fhwa-stol/carma-platform/issues/1947

## Related Issue
https://github.com/usdot-fhwa-stol/carma-platform/issues/1932
https://github.com/usdot-fhwa-stol/autoware.ai/pull/232

## Motivation and Context
TSMO UC3 incremental integration testing

## How Has This Been Tested?
Integration tested on Black Pacifica with 3 scenarios:
- speeding to make the green light
- slowing down to make the green light
- cruising at minimum speed to stop at red light, before passing at green

## Types of changes

- [X] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [X] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: carma_wm/test/CARMAWorldModelTest.cpp [hunk 1]
```diff
@@ -1291,7 +1291,7 @@ TEST(CARMAWorldModelTest, sampleRoutePoints)
   }
 }
 
-TEST(CARMAWorldModelTest, getTrafficSignalId)
+TEST(CARMAWorldModelTest, DISABLED_getTrafficSignalId)
```
Comment (msmcconnell) [inline carma_wm/test/CARMAWorldModelTest.cpp:1294]: Why are these unit tests disabled?

File: light_controlled_intersection_tactical_plugin/src/light_controlled_intersection_tactical_plugin.cpp [hunk 2]
```diff
@@ -63,7 +63,7 @@ namespace light_controlled_intersection_tactical_plugin
         
         if (current_lanelets.empty())
         {
-            RCLCPP_ERROR_STREAM(logger_->get_logger(), "Given vehicle position is not on the road! Returning...");
+            RCLCPP_DEBUG_STREAM(logger_->get_logger(), "Given vehicle position is not on the road! Returning...");
```
Comment (msmcconnell) [inline light_controlled_intersection_tactical_plugin/src/light_controlled_intersection_tactical_plugin.cpp:66]: Why debug not error?

**Review by msmcconnell — CHANGES_REQUESTED** [review]
There's a lot of near term TODOs here. We need to track these. Please open a github issue that describes where they all are so they can be fixed.

File: carma_wm/test/CARMAWorldModelTest.cpp (hunk 1 above)
Comment (MishkaMN) [reply carma_wm/test/CARMAWorldModelTest.cpp]: accidentally pushed before enabling back

File: light_controlled_intersection_tactical_plugin/src/light_controlled_intersection_tactical_plugin.cpp (hunk 2 above)
Comment (MishkaMN) [reply light_controlled_intersection_tactical_plugin/src/light_controlled_intersection_tactical_plugin.cpp:66]: made it back to error

**Review by msmcconnell — APPROVED** [review]

**Pull request merged**